.github/scripts/generate_binary_build_matrix.py

```diff
@@ -54,33 +58,57 @@
 
 
 def is_pull_request():
-    return os.environ.get("GITHUB_HEAD_REF")
+    return False
+    # return os.environ.get("GITHUB_HEAD_REF")
 
-def generate_matrix():
+def generate_matrix(include_cuda=True, include_rocm=False):
     python_versions = FULL_PYTHON_VERSIONS
-    arches = FULL_ARCHES
+    cuda_versions = CUDA_ARCHES
+    rocm_versions = ROCM_ARCHES
+    arches = ["cpu"]
     if is_pull_request():
         python_versions = [python_versions[-1]]
-        arches = ["cpu", CUDA_ARCHES[-1], ROCM_ARCHES[-1]]
+        cuda_versions = [cuda_versions[-1]]
+        rocm_versions = [rocm_versions[-1]]
+    if include_cuda:
+        arches += cuda_versions
+    if include_rocm:
+        arches += rocm_versions
     matrix = []
     for item in itertools.product(python_versions, arches):
         python_version, arch_version = item
         # Not my favorite code here
         gpu_arch_type = "cuda"
-        if "rocm" in CONTAINER_IMAGES[arch_version]:
+        if "rocm" in WHEEL_CONTAINER_IMAGES[arch_version]:
```

Comment:
this seems to assume the same arch_version exists on both conda and wheel image dict keyset. yes?

Reply:
Unfortunately not necessarily, conda does not have a rocm equivalent just yet

Reply:
oops. my mistake. I meant wheel container dict keyset will be a superset of conda container dict keyset